### PR TITLE
add hands-on time series notebook

### PR DESCRIPTION
**Pull request by FlorentijnD**
This PR adds a hands-on notebook to get familiar with time series data processing. 
The notebook covers the following aspects:

- Data transformation : go from a tabular data set to a time series friendly data set using pandas.
- Data imputation : impute missing values using a number of built-in time series imputation approaches and visualise.
- Decomposition : perform data decomposition to extract the trend, seasonality component and the residuale from the time series dataset using STL.
- Stationarity check : check whether your data has a constant mean, variance and covariance over time using KPSS and ADF.
- [Extra] Outlier detection : Identify data points that could be erroneous using linear regression, SVM's or isolated forests.

This notebook was developed for the purpose of having a hands-on session during the chapter conference talk about time series in the first quarter of 2021.

### CODE REVIEW

**Comment by Fematich** [comment]
Thx @FlorentijnD!

I would rephrase the title & some content a bit to make it easier digestable for non-ML6ers.
The goal of this repo is to have an open mindset & also make it directly externally visible and useful. I think it still a good match for the external quick-tips, but would need some updates:
1) I would rename sd towards "structured_data", natural_language_processing" would be unnecessarily long, however I believe SD as abbreviation is not that generally interpretable.
2) I would flatten and rename the folder/name as well (given that CC has no meaning externally), and the title on itself should be self-explanatory of what it contains, following the stucture under [NLP](https://github.com/ml6team/quick-tips/tree/main/nlp):
 - 2021_02_08_timeseries_getting_started
3) Add a small Readme under s"tructured_data" folder with short description & link to the notebook (see [NLP Readme](https://github.com/ml6team/quick-tips/tree/main/nlp))
4) The notebook also still contains the 'exercise format', I would remove this 'your_answer_here' parts as well


A lot of text, but it's only regarding formatting & basically making sure that it's useful outside of ML6 as well (given that it's an external repo).

Thanks for the great notebook!
Matthias

**Comment by FlorentijnD** [comment]
> Thx @FlorentijnD!
> 
> I would rephrase the title & some content a bit to make it easier digestable for non-ML6ers.
> The goal of this repo is to have an open mindset & also make it directly externally visible and useful. I think it still a good match for the external quick-tips, but would need some updates:
> 
> 1. I would rename sd towards "structured_data", natural_language_processing" would be unnecessarily long, however I believe SD as abbreviation is not that generally interpretable.
> 2. I would flatten and rename the folder/name as well (given that CC has no meaning externally), and the title on itself should be self-explanatory of what it contains, following the stucture under [NLP](https://github.com/ml6team/quick-tips/tree/main/nlp):
> 
> * 2021_02_08_timeseries_getting_started
> 
> 1. Add a small Readme under s"tructured_data" folder with short description & link to the notebook (see [NLP Readme](https://github.com/ml6team/quick-tips/tree/main/nlp))
> 2. The notebook also still contains the 'exercise format', I would remove this 'your_answer_here' parts as well
> 
> A lot of text, but it's only regarding formatting & basically making sure that it's useful outside of ML6 as well (given that it's an external repo).
> 
> Thanks for the great notebook!
> Matthias

Thanks for the remarks! Should be solved now, except for the Readme @julienschuermans

**Comment by julienschuermans** [comment]
Thanks @FlorentijnD, looking great!
I've added some final touches:

* added a README for the `structured_data` directory.
* added a README specifically for your tip.
* renamed the notebook.
* removed/renamed pointers to previous 'exercises' and 'spoilers' in the notebook

**Review by julienschuermans — APPROVED** [review]
looks good to me!

**Review by TDehaene — APPROVED** [review]
Looking good :-), welcome to the quick-tip family.

Personally, I'm a fan of clearing some non-relevant output of notebooks:

- graphs and pandas heads are useful
- pip install outputs are less relevant and can add a clutter-y look

**Comment by FlorentijnD** [comment]
@Fematich WFA

**Review by Fematich — APPROVED** [review]
Perfect, LGTM thanks for making this happen :-)!
